Bottom: df4460d78f61e2b742a336d7f0bd69e2473f4510
Top:    c50ad436d83038f2d2b263b932641a68f09e44b2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-23 17:52:31 +0000

Refresh of remove-draw-call-from-resetfou

---

diff --git a/src/action.c b/src/action.c
index fe28c84..997e29e 100644
--- a/src/action.c
+++ b/src/action.c
@@ -2365,17 +2365,26 @@ ActionConnection (int argc, char **argv, int x, int y)
 
 	case F_ResetLinesAndPolygons:
 	  if (ResetFoundLinesAndPolygons (true, true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	  break;
 
 	case F_ResetPinsViasAndPads:
 	  if (ResetFoundPinsViasAndPads (true, true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	  break;
 
 	case F_Reset:
 	  if (ResetConnections (true, true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	  break;
 	}
       RestoreCrosshair (true);
@@ -2789,7 +2798,10 @@ ActionDisplay (int argc, char **argv, int childX, int childY)
 	  if (TEST_FLAG (AUTODRCFLAG, PCB) && Settings.Mode == LINE_MODE)
 	    {
 	      if (ResetConnections (true, true))
-		IncrementUndoSerialNumber ();
+		{
+		  IncrementUndoSerialNumber ();
+		  Draw ();
+		}
 	      if (Crosshair.AttachedLine.State != STATE_FIRST)
 		LookupConnection (Crosshair.AttachedLine.Point1.X,
 				  Crosshair.AttachedLine.Point1.Y, true, 1,
diff --git a/src/find.c b/src/find.c
index 1c9e94a..696d510 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3490,11 +3490,7 @@ ResetFoundPinsViasAndPads (bool save_undo, bool redraw)
   }
   END_LOOP;
   if (change)
-    {
-      SetChangedFlag (true);
-      if (redraw)
-        Draw ();
-    }
+    SetChangedFlag (true);
   return change;
 }
 
@@ -3559,11 +3555,7 @@ ResetFoundLinesAndPolygons (bool save_undo, bool redraw)
   }
   ENDALL_LOOP;
   if (change)
-    {
-      SetChangedFlag (true);
-      if (redraw)
-        Draw ();
-    }
+    SetChangedFlag (true);
   return change;
 }
 
@@ -3926,7 +3918,10 @@ DRCAll (void)
   TheFlag = FOUNDFLAG | DRCFLAG | SELECTEDFLAG;
 
   if (ResetConnections (true, true))
-    IncrementUndoSerialNumber ();
+    {
+      IncrementUndoSerialNumber ();
+      Draw (); /* XXX: Not sure if this is required */
+    }
 
   User = false;
 
diff --git a/src/report.c b/src/report.c
index da4b600..486ebf4 100644
--- a/src/report.c
+++ b/src/report.c
@@ -689,7 +689,10 @@ ReportAllNetLengths (int argc, char **argv, int x, int y)
 
     got_one:
       if (ResetConnections (true, true))
-        IncrementUndoSerialNumber ();
+	{
+	  IncrementUndoSerialNumber ();
+	  Draw ();
+	}
       length = XYtoNetLength (x, y, &found);
 
       gui->log("Net %s length %.*f %s\n", netname, prec, length*scale, units_name);
@@ -705,7 +708,10 @@ ReportNetLength (int argc, char **argv, int x, int y)
   int found = 0;
 
   if (ResetConnections (true, true))
-    IncrementUndoSerialNumber ();
+    {
+      IncrementUndoSerialNumber ();
+      Draw ();
+    }
   gui->get_coords ("Click on a connection", &x, &y);
 
   length = XYtoNetLength (x, y, &found);
diff --git a/src/set.c b/src/set.c
index 837b786..84c37f5 100644
--- a/src/set.c
+++ b/src/set.c
@@ -293,7 +293,10 @@ SetMode (int Mode)
       if (Mode == LINE_MODE && TEST_FLAG (AUTODRCFLAG, PCB))
 	{
 	  if (ResetConnections (true, true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	}
     }
